src/raster/rasterband.rs

```diff
@@ -12,6 +15,68 @@ use ndarray::Array2;
 
 use crate::errors::*;
 
+/// Extra options used to read a raster.
+///
+/// For documentation, see `gdal_sys::GDALRasterIOExtraArg`.
+#[derive(Debug)]
+pub struct RasterIOExtraArg {
+    pub n_version: libc::c_int,
+    pub e_resample_alg: GDALRIOResampleAlg::Type,
```

Comment:
i think we should have an enum for the resample algorithms. That will make it easier for most users.

Reply:
`GDALRIOResampleAlg::Type` are already constants. Should I create a mod inside rasterband to map it?

Reply:
@jdroenner I added an enum with the names of the algorithms and a function to map them to integers (GDAL uses integers).